frontend/src/metabase-lib/parameters/utils/template-tags.ts

```diff
@@ -77,10 +86,14 @@ export function getParametersFromCard(
 
   if (card.parameters && !_.isEmpty(card.parameters)) {
     return card.parameters;
+  } else {
+    return getTemplateTagParametersFromCard(card);
   }
+}
 
+export function getTemplateTagParametersFromCard(card: Card) {
   const tags = getTemplateTagsForParameters(card);
-  return getTemplateTagParameters(tags);
+  return getTemplateTagParameters(tags, card.parameters);
```

Comment:
nit: I was like staring at this line for minutes wondering why we need to call the same function twice. But they're different functions with **too** similar names 🤦 . No immediate action here, but I think the name could be improved to reduce further confusion.